Sources/AblyChat/AblyCocoaExtensions/Ably+Concurrency.swift

```diff
@@ -0,0 +1,30 @@
+import Ably
+
+// This file contains extensions to ably-cocoa’s types, to make them easier to use in Swift concurrency.
+// TODO: remove once we improve this experience in ably-cocoa (https://github.com/ably/ably-cocoa/issues/1967)
+
+internal extension ARTRealtimeChannelProtocol {
+    func attachAsync() async throws {
+        try await withCheckedThrowingContinuation { (continuation: CheckedContinuation<Void, Swift.Error>) in
```

Comment:
`{ continuation in` looks cleaner I think?

Reply:
You mean without the type? I _think_ that it needs the type to compile, because there's no way for it to infer the `Void`.

Reply:
I replaced the `Swift.Error` with a `_` to let the compiler infer it (forgot this feature existed), and did the same with some existing code too.